### DISCUSSION TITLE
Exported GLB is too large

### DISCUSSION THREAD

**Discussion by MenachemBerkovich** [CadQuery/cadquery#1625] [Q&A]
I have simple script to load some step file, and export it into glb file.

```
file = cadquery.importers.importStep("E:\Inabbeet\Front 315_360 M9967.stp")
final_assembly = cadquery.Assembly(file)
final_assembly.save(path="expGlb.glb", exportType="GLB")
```
But the exported file is about one GB (!!!) While source step file is 77.2 MB only.
Why?
Thank you.

**Comment by jmwright** [comment]
It is hard to say without seeing the STEP file. However, glTF/glb is a mesh-based format, and for a complex shape the mesh can get fairly large.  You could try adjusting the `tolerance` and `angularTolerance` parameters to see if you can decrease the size.

https://github.com/CadQuery/cadquery/blob/master/cadquery/assembly.py#L468

**Comment by Wildfirefox** [comment]
Hi everybody, im a bit late to the party. We faced the same issue and have no simple solution yet.
We implemented a python service which does the conversion from colored STEP files to colored GLB's based on the OCP stubs. Were about to "reimplement" the logic with the cadquery base API which should be not that difficult with some caveats.

@MenachemBerkovich  When i came to the same line of code, i noticed that the GLB files were significant larger than our export based on the OCP stubs. In our first implementation we set the mergeFaces flag to true on the RWGltf_CafWriter, which gaves us the smaller glb files.

OCCT API call for setting up the writer:
![image](https://github.com/user-attachments/assets/53610467-003c-480c-9266-5f7e1c57a226)

cadquery implementation of the writer setup:
![image](https://github.com/user-attachments/assets/8dd4a112-41c5-41b3-82d8-04128221c322)

So we noticed, that there is no parameter for forcing the writer to call SetMergeFaces true on the cadquery implementation. This could help to decrease the file sizes of GLTF/GLB if desired. 

@jmwright Wouldn't it be good to provide a additonal boolean parameter specifc for the GLTF/GLB conversion? As i understand there are already type specific parameters in the save method:
![image](https://github.com/user-attachments/assets/b7b59612-45e4-4ff1-9fb5-de192a4c2b35)

Or am i missing something to "enable" the writer on the cadquery site to get to the desired behaviour? Were also in the need to get smaller files, this could be a nice feature 😊
